include/quicr/track_name.h

```diff
@@ -192,29 +214,44 @@ namespace quicr {
 
         friend bool operator>=(const TrackNamespace& lhs, const TrackNamespace& rhs) noexcept { return !(lhs < rhs); }
 
-        bool Contains(const TrackNamespace& other) const noexcept
+        bool IsPrefixOf(const TrackNamespace& other) const noexcept
         {
             if (this->size() > other.size()) {
                 return false;
             }
 
-            auto other_view = Span{ other }.subspan(0, this->size());
-            return std::equal(this->begin(), this->end(), other_view.begin(), other_view.end());
+            return this->hash() == other.hash(this->hash_.size());
+        }
+
+        bool HasSamePrefix(const TrackNamespace& other) const noexcept
+        {
+            const std::size_t min_size = std::min(this->hash_.size(), other.hash_.size());
+            return this->hash(min_size) == other.hash(min_size);
+        }
+
+      private:
+        std::size_t hash(std::size_t offset = -1) const noexcept
+        {
+            std::size_t value = 0;
+            for (const auto& h : Span{ hash_ }.subspan(0, std::min(offset, hash_.size()))) {
+                hash_combine(value, h);
+            }
+            return value;
         }
 
       private:
         std::vector<uint8_t> bytes_;
         std::vector<Span<const uint8_t>> entries_;
+        std::vector<std::size_t> hash_;
```

Comment:
Seems odd to be singular. Maybe `hashes_` is a better name considering this is a vector of hashes of the namespace tuple.